airflow/models/dagrun.py

```diff
@@ -272,11 +272,75 @@ def get_state(self):
         return self._state
 
     def set_state(self, state: DagRunState) -> None:
+        """Change the state of the DagRan.
```

Comment:
`set_state` is only used in scheduler & backfill jobs. I don't see how your changes connect to manually marking dagrun success or failed.

Reply:
> `set_state` is only used in scheduler & backfill jobs. I don't see how your changes connect to manually marking dagrun success or failed.

We use property with `set_state` [here](https://github.com/apache/airflow/blob/fd9d0b0f7de515b661ec459ba66778dc9730cf0f/airflow/models/dagrun.py#L347-L349). And we modify the state [here](https://github.com/apache/airflow/blob/fd9d0b0f7de515b661ec459ba66778dc9730cf0f/airflow/api/common/mark_tasks.py#L368).